### PR TITLE
Added distributed iterator

### PR DESCRIPTION
**Pull request by Vorotyntsev**
Added distributed iterator for lists, sets, sorted sets and scored sorted sets. 
List iterator is based on `LRANGE` command, other - on `SSCAN` and `ZSCAN`

General algorithm is following:
1. Retrieve saved cursor
2. If it doesn't exist, use 0 as default
3. If cursor is -1, it's exhausted and empty result should be returned
4. Scan next `count` elements
5. Update cursor
6. For multimaps filter elements by ttl

### CODE REVIEW

**Review by mrniko — CHANGES_REQUESTED** [review]
Looks great!

But please reformat if ~ else ~ end statements according to the format used in other Lua-scripts;

```
if ... then
   ...
else 
   ...
end;
```

File: redisson/src/main/java/org/redisson/RedissonList.java [hunk 1]
```diff
@@ -300,7 +302,55 @@ public void clear() {
...
+            }
+        };
+    }
+
+    private ScanResult<Object> distributedScanIterator(String iteratorName, int count) {
+        return get(distributedScanIteratorAsync(iteratorName, count));
+    }
+
+    private RFuture<ScanResult<Object>> distributedScanIteratorAsync(String iteratorName, int count) {
+        List<Object> args = new ArrayList<>(1);
```
Comment (mrniko) [inline redisson/src/main/java/org/redisson/RedissonList.java:333]: don't use list if arguments are static

File: redisson/src/main/java/org/redisson/RedissonListMultimapValues.java [hunk 2]
```diff
@@ -363,7 +365,68 @@ public void clear() {
...
+            }
+        };
+    }
+
+    private ScanResult<Object> distributedScanIterator(String iteratorName, int count) {
+        return get(distributedScanIteratorAsync(iteratorName, count));
+    }
+
+    private RFuture<ScanResult<Object>> distributedScanIteratorAsync(String iteratorName, int count) {
+        List<Object> args = new ArrayList<>(2);
```
Comment (mrniko) [inline redisson/src/main/java/org/redisson/RedissonListMultimapValues.java:396]: don't use list if arguments are static

File: redisson/src/main/java/org/redisson/RedissonSortedSet.java [hunk 3]
```diff
@@ -387,7 +391,55 @@ public boolean trySetComparator(Comparator<? super V> comparator) {
...
+            }
+        };
+    }
+
+    private ScanResult<Object> distributedScanIterator(String iteratorName, int count) {
+        return get(distributedScanIteratorAsync(iteratorName, count));
+    }
+
+    private RFuture<ScanResult<Object>> distributedScanIteratorAsync(String iteratorName, int count) {
+        List<Object> args = new ArrayList<>(1);
```
Comment (mrniko) [inline redisson/src/main/java/org/redisson/RedissonSortedSet.java:422]: don't use list if arguments are static

**Review by mrniko — APPROVED** [review]

**Comment by mrniko** [comment]
Thanks for contribution!

**Comment by Vorotyntsev** [comment]
Glad to help. Thanks!

**Pull request merged**